internal/telemetry/message.go

```diff
@@ -5,8 +5,20 @@
 
 package telemetry
 
-// Request is the common high-level structure encapsulating a telemetry request
+import "net/http"
+
+// Request captures all necessary information for a telemetry event submission
+// so we do not need to read directly from our telemetry client when submitting
+// asynchronously
```

Comment:
I thought I left a comment here before, but it looks like it didn't go through: was reading from the client during request submission causing problems before? I'm not necessarily opposed to this, just want to make sure I understand the motivation.

Reply:
Was part of the client being modified during submission?

Reply:
Looking more into it I can't find a place where submission modifies the client (nor a place during submission where a part of the client is being read that might be modified elsewhere). 

Previously, I had code [like this](https://github.com/DataDog/dd-trace-go/blob/ed83c410fa72722271367afc1bb92ac67b95d420/internal/telemetry/client.go#LL445-L465C2) which caused the unit [tests to fail](https://github.com/DataDog/dd-trace-go/actions/runs/4306399835/jobs/7510152445) due to race conditions, which motivated me to separate the submission process from the client to avoid any issues reading from client fields that might be modified. That logic is removed now though.

Would you be fine with keeping it like this for now? `submit` would be slightly cleaner since the header field is already populated, but we may need to consider 'reconnecting' the request with the telemetry client in the future (e.g. if we want to support something like saving failed requests in memory and then retrying at a later flush interval).

Reply:
Okay, I understand where the issue came from. In the interest of making progress on this PR, I'm fine with keeping this the way it is for now. I think we can simplify this in the future so Request does one thing (hold the body of a request).